source/FAST/Data/ImagePyramid.cpp

```diff
@@ -32,7 +32,7 @@ void ImagePyramid::create(int width, int height, int channels, int levels) {
 		currentWidth = width / std::pow(2, currentLevel);
 		currentHeight = height / std::pow(2, currentLevel);
 
-        if(currentWidth < 4096 || currentHeight < 4096)
+        if(currentWidth < 1024 || currentHeight < 1024)  // lowered the limit from 4096 to 1024
```

Comment:
This may affect performance

Reply:
Yes, but for the same reason above, it seemed necessary for the specific use case mentioned. The image plane I was testing on had dimensions in the order of 2k x 4k.